xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/api/Document.java

```diff
@@ -3247,4 +3258,33 @@ public int getLocalReferenceMaxLength()
     {
         return this.doc.getLocalReferenceMaxLength();
     }
+
+    /**
+     * Define the author of the document to be displayed, without changing the metadata and content authors.
+     * @param displayedAuthor the author to be displayed.
+     * @since 14.0RC1
+     */
+    @Unstable
+    public void setDisplayedAuthor(UserReference displayedAuthor)
+    {
+        DefaultDocumentAuthors documentAuthors;
+        if (this.doc.getAuthors() != null) {
```

Comment:
I'm really not a fan of how complex it is to set authors in a XWikiDocument.

Reply:
Actually this check should not be needed since I also check it in the setter. Will remove it.

Reply:
My comment is actually more for the fact that you have to create a new DefaultDocumentAuthors (which is an internal class which does not make this whole thing complex but simply impossible for an extension as I already indicated in a previous comment), duplicate the current one except the one value you want to change and set it.

Honestly, I don't see why DocumentAuthors does not simply have setters.

Reply:
Fixed by adding setters in DocumentAuthors.